doc/devel/style_guide.rst

```diff
@@ -155,14 +155,18 @@ reliability and consistency in documentation. They are not interchangeable.
   |                  |                          |   rotational |              |
   |                  |                          |   motion."   |              |
   +------------------+--------------------------+--------------+--------------+
-  | Explicit,        | Explicit approach of     | - Explicit   | - object     |
-  | Object Oriented  | programming in           | - explicit   |   oriented   |
-  | Programming (OOP)| Matplotlib.              | - OOP        | - OO-style   |
+  | Axes interface   | Usage pattern in which   | - Axes       | - explicit   |
```

Comment:
I feel like the third column misses the need to use `Figure` objects too - maybe add "cal methods on the Figure object" as a third correct terminology?

There's also `Axis` objects... do we think it's safe to leave these off here, because one (well at least I...) always get an `Axis` directly from `Axes.{x,y}axis`?

Reply:
Changed the second column to

> Usage pattern in which one calls methods on Axes and Figure (and sometimes other Artist) objects to configure the plot.

to cover the concepts.

The third column are examples for correct use. I'm sure people can generalize from that.